recipe-client-addon/lib/TelemetryEvents.jsm

```diff
@@ -0,0 +1,61 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+"use strict";
+
+const {utils: Cu, interfaces: Ci} = Components;
+Cu.import("resource://gre/modules/Services.jsm");
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+
+XPCOMUtils.defineLazyModuleGetter(
+  this, "CleanupManager", "resource://shield-recipe-client/lib/CleanupManager.jsm"
+);
+XPCOMUtils.defineLazyModuleGetter(
+  this, "LogManager", "resource://shield-recipe-client/lib/LogManager.jsm"
+);
+
+this.EXPORTED_SYMBOLS = ["TelemetryEvents"];
+
+const TELEMETRY_CATEGORY = "normandy";
+const log = LogManager.getLogger("preference-experiments");
+
+let isStarted = false;
+
+const TelemetryEvents = {
+  startup() {
+    Services.telemetry.registerEvents(TELEMETRY_CATEGORY, {
+      enroll: {
+        methods: ["enroll"],
+        objects: ["preference-study", "addon-study"],
+        extra_keys: ["experimentType", "branch", "addonId", "addonVersion"],
+        record_on_release: true,
+      },
+
+      unenroll: {
+        methods: ["unenroll"],
+        objects: ["preference-study", "addon-study"],
+        extra_keys: ["reason", "resetValue"],
+        record_on_release: true,
+      },
+    });
+
+    Services.telemetry.setEventRecordingEnabled(TELEMETRY_CATEGORY, true);
```

Comment:
You don't need this for events that are registered at runtime through `registerEvents()`.
This is only needed for events shipping in Firefox code (to avoid collecting all the time).